File: back/src/main/java/com/triptrace/domain/trip/trip/dto/TripSearchCondition.java
package com.triptrace.domain.trip.trip.dto;

import java.util.List;

// 검색 조건 생성
public record TripSearchCondition (
    List<String> tokens,
    TripSearchScope scope,
    String country,
    String city,
    TripSearchSort sort
    ){
    public  boolean hasKeyword(){
        return tokens != null && !tokens.isEmpty();
    }
}

File: back/src/main/java/com/triptrace/domain/trip/trip/dto/TripSearchCondition.kt
package com.triptrace.domain.trip.trip.dto

// 검색 조건 생성
@JvmRecord
data class TripSearchCondition(
    val tokens: List<String>,
    val scope: TripSearchScope,
    val country: String?,
    val city: String?,
    val sort: TripSearchSort
) {
    fun hasKeyword(): Boolean = tokens.isNotEmpty()
}
